datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -327,7 +327,7 @@ async fn simple_scalar_function_pow() -> Result<()> {
 
 #[tokio::test]
 async fn simple_scalar_function_substr() -> Result<()> {
-    roundtrip("SELECT * FROM data WHERE a = SUBSTR('datafusion', 0, 3)").await
+    roundtrip("SELECT SUBSTR(f, 1, 3) FROM data").await
```

Comment:
is there a reason to change the test?

Maybe we could add this particular query as an additioanl query (to show the existing behavior is not changed). Something like

```suggestion
    roundtrip("SELECT * FROM data WHERE a = SUBSTR('datafusion', 0, 3)").await
    roundtrip("SELECT SUBSTR(f, 1, 3) FROM data").await
```

Reply:
> is there a reason to change the test?

Yes - the original query gets optimized by DF into `SELECT * FROM data WHERE a = "dat"` before being converted into Substrait, i.e. the whole SUBSTRING call is optimized away:
```
Filter: CAST(data.a AS Utf8) = Utf8("da")
  TableScan: data projection=[a, b, c, d, e, f, g], partial_filters=[CAST(data.a AS Utf8) = Utf8("da")]
```

> Maybe we could add this particular query as an additioanl query (to show the existing behavior is not changed). 

I can add it back, but it doesn't really test what it tries to test 😅 given that would you still like to have it?

Reply:
Thank you for the explanation

No need to change the PR

The issue seems to be that DataFusion partially evaluates the expression